File: Source/WickedSick.Server.XamlParser/Elements/Media/VSM/VisualTransition.cs
using WickedSick.Server.XamlParser.Elements.Media.Animation;
using WickedSick.Server.XamlParser.Elements.Types;

namespace WickedSick.Server.XamlParser.Elements.Media.VSM
{
    public class VisualTransition : DependencyObject
    {
        public static readonly PropertyDescription StoryboardProperty = PropertyDescription.Register("Storyboard", typeof(Storyboard), typeof(VisualTransition), true);
        public static readonly PropertyDescription FromProperty = PropertyDescription.Register("From", typeof(string), typeof(VisualTransition));
        public static readonly PropertyDescription ToProperty = PropertyDescription.Register("To", typeof(string), typeof(VisualTransition));
        public static readonly PropertyDescription GeneratedDurationProperty = PropertyDescription.Register("GeneratedDuration", typeof(Duration), typeof(VisualTransition));
    }
}

File: src/Media/VSM/VisualTransition.ts
/// <reference path="../../Core/DependencyObject.ts" />

module Fayde.Media.VSM {
    export class VisualTransition extends DependencyObject {
        From: string = null;
        To: string = null;
        Storyboard: Animation.Storyboard;

        private _GeneratedDuration: Duration = null;
        get GeneratedDuration(): Duration { return this._GeneratedDuration; }
        set GeneratedDuration(value: Duration) { this._GeneratedDuration = Fayde.ConvertAnyToType(value, Duration); }

        DynamicStoryboardCompleted: boolean = true;
        ExplicitStoryboardCompleted: boolean = true;
        GeneratedEasingFunction: Animation.EasingFunctionBase;
        get IsDefault(): boolean { return this.From == null && this.To == null; }
    }
    Fayde.RegisterType(VisualTransition, "Fayde.Media.VSM", Fayde.XMLNS);
}